### ISSUE TITLE
Silencing reframe's output also silences any other file handler

### ISSUE THREAD

**Issue by vkarak** [reframe-hpc/reframe#2784]
## How to reproduce

Add this to your config:

```python
            'handlers': [
                {
                    'type': 'file',
                    'name': 'reframe.out',
                    'level': 'info',
                    'format': '%(message)s'
                }
            ],
```

Then run reframe with `-qq`. Instead of producing the normal output in `reframe.out`, this is empty. If you remove the `-qq` options, then you get your output in `reframe.out` as expected.

## The cause

The problem comes (most likely) from the following part of the code:

https://github.com/reframe-hpc/reframe/blob/6c3d163188574a882f509b905b5752c6ee6795e9/reframe/core/logging.py#L758-L759

called by the `adjust_verbosity()` logger's method. The problem is that a `FileHandler` _is_ a `StreamHandler` in [Python terms](https://docs.python.org/3/library/logging.handlers.html#filehandler), therefore it is also silenced. The intent here was to silence only the stream handlers to suppress the output.

**Comment by teojgo** [comment]
@vkarak I think this should have been fixed by #2786 right?

**Comment by vkarak** [comment]
Yes, but since the main branch is not `master` anymore, this is not closed automatically and since we haven't yet merged  `master` into `develop`, I just moved in the Kanban board (check status) so that we close it once we update `develop`.
